internal/backend/auth/auth.go

```diff
@@ -0,0 +1,50 @@
+package auth
+
+import (
+	"net/http"
+
+	"go.uber.org/zap"
+)
+
+type ProviderDetails struct {
+	Name   string
+	Config map[string]string
+}
+
+type Authenticator interface {
+	Middleware(next http.Handler) http.Handler
+	Provider() ProviderDetails
+}
+
+const AuthenticatedUserCtxKey = "auth:user_details"
```

Comment:
See lint error: define this key as a `struct{}` instead of a string to prevent collisions.

Reply:
fixed